src/libraries/System.Private.Xml/src/System/Xml/Serialization/ReflectionXmlSerializationWriter.cs

```diff
@@ -1398,5 +1402,61 @@ public static MemberInfo GetMember(Type declaringType, string memberName)
 
             return memberInfo;
         }
+
+        [RequiresUnreferencedCode(XmlSerializer.TrimSerializationWarning)]
+        public static MemberInfo GetEffectiveGetInfo(Type declaringType, string memberName)
+        {
+            MemberInfo memberInfo = GetMember(declaringType, memberName, true)!;
+
+            // For properties, we might have a PropertyInfo that does not have a valid
+            // getter at this level of inheritance. If that's the case, we need to look
+            // up the chain to find the right PropertyInfo for the getter.
+            if (memberInfo is PropertyInfo propInfo && propInfo.GetMethod == null)
+            {
```

Comment:
What happens if a base type has a property called Foo and the derived type has a method called Foo? GetMember would return the MethodInfo for the dervived Foo method and this code won't search the base type for the property.

Reply:
Crikey! But as we've talked about before, 'hidden' members are a mess in the serializers right now. This would fall under the broader work we need to do in #53051.